misaligned.c

```diff
@@ -16,6 +31,16 @@ int printColorMap() {
 int main() {
     int result = printColorMap();
     assert(result == 25);
+    printf("Pair number  1 : %-6s | %-6s\n", majorColorlist[0], minorColorlist[0]);
+    printf("Pair number  4 : %-6s | %-6s\n", majorColorlist[3], minorColorlist[3]);
+    printf("Pair number 16 : %-6s | %-6s\n", majorColorlist[15], minorColorlist[15]);
+    const char* paironemajorcolor = "White";
+    const char* pairtwominorcolor = "Orange";
+    //Test cases to verify colorpair
+    assert(majorColorlist[0]==paironemajorcolor);
+    assert(minorColorlist[1]==pairtwominorcolor);

```

Comment:
if the intention is to express behavior of production code knowing that the production code has flaws, then this test code is fine. But in that case the production code should not have been updated. Only test cases  with asserts to describe the behavior should be written.

But if you look at the production code which is modified , then the test cases are not sufficient and indeed has a bug.

Also the test cases should have a mechanism to check the alignment of string output which is not done here.

Reply:
I have already added this 
//Test case to check string alignment
    assert(misalignederrorcount==0);
to check the alignment of string

Reply:
Production code is also updated already. Kindly let me know what is wrong exactly. Thank you!

Reply:
Sorry, I wasn't clear 🙂. The approach to check for misalignment is appreciated, but that does not catch all the bugs.
What I meant with bug was, minorColor indexing is done with 'i' , instead 'j' should have been used. Please check line no. 25. Now the question to ourself is whether there is a test code that catches this bug🙂

Also the check/assert of whether the production code is performing as expected , should be in test methods.
Just a way of doing it would be to maintain a list of all the output strings and verify against expected output which we define as a list of strings within test code and this is in the format what is expected. Then use this to compare against output list.
Let me know if things are not clear 👍.

Reply:
Sure , thanks for the explanation. Regarding the indexing I have already updated it properly in 
https://github.com/clean-code-craft-tcq-3/test-failer-in-c-ArchanaDeviSelvaraj/blob/ArchanaDeviSelvaraj-patch-1/misaligned.c 
this file.
I have added asserts like  
//Test cases to verify colorpair
    assert(majorColorlist[0]==paironemajorcolor);
    assert(minorColorlist[1]==pairtwominorcolor);
to find the bug related to indexing.
Please correct me if something is wrong.
I will also further work on regarding the comments you have provided.
Since I have updated in 2 different branches(one for fail on false positive assignment and other for pass the failing test assignment) it is creating confusion I guess. Hereafter I will try to maintain it properly.

Reply:
Yes, so there it is the source of confusion 😊.it's clear now. Anyways, good that we had the discussion 👍. The tests asserts might be extended to cover some intermediate pairs. But that's fine, as long as the intention is clearly understood , goal of the review is achieved 👍

Reply:
Thanks a lot for your quick feedback and hints😊